06-Template-Distribution-Protocol.md

```diff
@@ -65,12 +65,22 @@ The primary template-providing function. Note that the `coinbase_tx_outputs` byt
 | coinbase_tx_outputs_count   | U32            | The number of transaction outputs included in coinbase_tx_outputs     |
 +-----------------------------+----------------+-----------------------------------------------------------------------+
 | coinbase_tx_outputs         | B0_64K         | Bitcoin transaction outputs to be included as the last outputs in     |
-|                             |                | the coinbase transaction                                              |
+|                             |                | the coinbase transaction. An additional coinbase tx output (other than|
+|                             |                | the output which pays the fees + block subsidy), will most likely be  |
+|                             |                | the coinbase witness commitment output.                               |
```

Comment:
This is bit unclear to me. Are there any outputs except for these?
Is there any first output since we are talking about "last outputs"?

Reply:
I think I understand now.

the coinbase_tx_outputs are constructed from
bytes inserted by the client (pool) that have maximum size `coinbase_output_max_additional_size`.
Those two arrays are concatenated `<client-specified-outputs> || coinbase_tx_outputs`

This statement 'An additional coinbase tx output will most likely be the coinbase witness commitment output` is quite confusing. It sounds like there would be yet-another transaction output for witness stuff.

Reply:
yes